src/mesh/analysis.rs

```diff
@@ -7,40 +8,113 @@ use crate::convert::{cast_i32, cast_usize};
 
 use super::{Face, Mesh, OrientedEdge, UnorientedEdge};
 
-pub fn compute_bounding_sphere<'a, I>(meshes: I) -> (Point3<f32>, f32)
-where
-    I: IntoIterator<Item = &'a Mesh> + Clone,
-{
-    let centroid = compute_centroid(meshes.clone());
-    let mut max_distance_squared = 0.0;
-
-    for mesh in meshes {
-        for vertex in &mesh.vertices {
-            let distance_squared = na::distance_squared(&centroid, vertex);
-            if distance_squared > max_distance_squared {
-                max_distance_squared = distance_squared;
-            }
+/// World-aligned bounding box contains the entire given geometry and defines an
+/// envelope aligned to the world (euclidean) coordinate system.
+pub struct BoundingBox {
+    minimum_point: Point3<f32>,
+    maximum_point: Point3<f32>,
+}
+
+impl BoundingBox {
+    pub fn new(min_x: f32, min_y: f32, min_z: f32, max_x: f32, max_y: f32, max_z: f32) -> Self {
+        let (minimum_x, maximum_x) = if min_x < max_x {
+            (min_x, max_x)
+        } else {
+            (max_x, min_x)
+        };
+        let (minimum_y, maximum_y) = if min_y < max_y {
+            (min_y, max_y)
+        } else {
+            (max_y, min_y)
+        };
+        let (minimum_z, maximum_z) = if min_z < max_z {
+            (min_z, max_z)
+        } else {
+            (max_z, min_z)
+        };
+        BoundingBox {
+            minimum_point: Point3::new(minimum_x, minimum_y, minimum_z),
+            maximum_point: Point3::new(maximum_x, maximum_y, maximum_z),
         }
```

Comment:
Unless you are adamant on having this, I'd remove `BoundingBox::new` and inline the logic in `BoundingBox::from_extremes` (also, I'd rename `from_extremes` to `new`). `BoundingBox::from_meshes` does not require these checks to happen.

Or do you imagine we'll be using this constructor?